### DISCUSSION TITLE
[RFC]: Update test262 infrastructure

### DISCUSSION THREAD

**Discussion by magic-akari** [swc-project/swc#8353] [RFC]
The last update on [test262-parser-tests](https://github.com/tc39/test262-parser-tests) was two years ago, and there are many new language syntax features that lack test cases. Therefore, I propose using [test262](https://github.com/tc39/test262) directly as the test suite.

If this proposal is accepted, we need to make some modifications:

1. We need to parse the top-level comments in each test file to extract metadata that indicates whether the test case is in strict mode, module mode, involves early errors, or includes other relevant details.

2. It may be necessary to create a bot for automatically updating the test cases.

**Comment by kdy1** [comment]
I think we should do it. Thank you for the notice!

**Reply by Boshen** [reply]
I have a badly written test runner (with parts copied from Biome) for testing against test262, babel and TypeScript: https://github.com/oxc-project/oxc/tree/main/tasks/coverage

Perhaps the easiest approach is to copy what's relevant into swc.
